hack/ginkgo-e2e.sh

```diff
@@ -166,15 +166,29 @@ elif [[ "${E2E_TEST_DEBUG_TOOL}" == "gdb" ]]; then
   program=("gdb")
 fi
 
+debug_opt=()
+# Here are some options which will be helpful for debugging, for example, test hangs for some reason (https://github.com/kubernetes/kubernetes/issues/111086).
+
+# Unmask this line to circumvent cases where the code grabbing the stdout/stderr pipe
+# is not under the framework control and may cause hangs (https://github.com/onsi/ginkgo/issues/851)
+# this option will also help to reveal more information when the test hangs (https://github.com/onsi/ginkgo/issues/970).
+# debug_opt+=(--ginkgo.output-interceptor-mode=none)
+
+# Unmask this line in order to emits more output including GinkgoWriter contents.
+# debug_opt+=(--ginkgo.v)
+
+# Unmask this line if you want the default reporter prints out captured output of passed tests.
+# debug_opt+=(--ginkgo.always-emit-ginkgo-writer)
+
 # NOTE: Ginkgo's default timeout has been reduced from 24h to 1h in V2, set it manually here as "24h"
 # for backward compatibility purpose.
-# Set --output-interceptor-mode= none to circumvent cases where the code grabbing the stdout/stderr pipe
-# is not under the framework control and may cause hangs (https://github.com/onsi/ginkgo/issues/851)
 "${program[@]}" "${ginkgo_args[@]:+${ginkgo_args[@]}}" "${e2e_test}" -- \
   "${auth_config[@]:+${auth_config[@]}}" \
+  "${debug_opt[@]:+${debug_opt[@]}}" \
   --ginkgo.flake-attempts="${FLAKE_ATTEMPTS}" \
+  --ginkgo.json-report="${GINKGO_JSON_REPORT:-}" \
```

Comment:
is this flag with an empty value ignored?

Reply:
yes, I tested this on my desktop.

Reply:
Is there a reason not to just place this under E2E_REPORT_DIR ?

Reply:
honestly i'd recommend just turning it on and not worrying about making it configurable.  i'd only hesitate if y'all have issues around disk space usage for CI - i imagine the reports will be sizeable!

Reply:
> Is there a reason not to just place this under E2E_REPORT_DIR ?

No reason actually, I just haven't noticed that dir. :)

Reply:
> honestly i'd recommend just turning it on and not worrying about making it configurable

by switching to `E2E_REPORT_DIR`, the file will be available as long as this report dir is set, I think this should be okay enough, we cannot hardcode it to a directory where we don't how to get it. :)

Reply:
hey - sorry, you might run into problems because of this issue:

https://github.com/onsi/ginkgo/issues/1001

let me know if you do

Reply:
no, I tried both junit and json, absolute paths works for me.